base_edifact/models/edifact.py

```diff
@@ -0,0 +1,364 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 PlanetaTIC - Marc Poch <mpoch@planetatic.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, api
+from datetime import datetime
+from slugify import slugify
+
+
+class BaseEdifact(models.AbstractModel):
+    _name = 'base.edifact'
+    _description =\
+        'Common methods to generate and parse Edifact plain text files'
+
+    def get_vat(self, partner):
+        vat = ''
+        if partner.vat:
+            vat = partner._split_vat(partner.vat)[1]
+        return vat
+
+    def get_partner_data(self, partner):
+        name = slugify(partner.name, separator=' ', lowercase=False,
+                       max_length=70)
+        city = slugify(partner.name, separator=' ', lowercase=False,
+                       max_length=35)
+        street = partner.street or ''
+        if partner.street2:
+            if street:
+                street += ' '
+            street += partner.street2
+        # Only ASCII chars:
+        street = slugify(street, separator=' ', lowercase=False, max_length=70)
+        return (name, city, street)
+
+    @api.model
+    def edifact_date(self, date):
+        return '%s%s%s' % (date.year,
+                           '{0:02d}'.format(date.month),
+                           '{0:02d}'.format(date.day))
+
+    # Methods to generate: invoice
+    @api.model
+    def edifact_invoice_init(self):
+        return 'INVOIC_D_93A_UN_EAN007\n'
+
```

Comment:
Thanks for the initiative on this matter.
Don't you think method with `invoice` should move to your edifact module dedicated to invoice ?